src/main/java/sirius/biz/jdbc/MongoAndElasticDatabaseController.java

```diff
@@ -0,0 +1,141 @@
+    @Permission(TenantUserManager.PERMISSION_SYSTEM_ADMINISTRATOR)
+    @Routed("/system/query")
+    public void db(WebContext webContext) {
+        String className = webContext.get("class").asString();
+        String queryString = webContext.get("query").asString();
+        int limit = webContext.get("limit").asInt(10);
+
+        StringBuilder sourceBuilder = new StringBuilder();
+        Counter counter = new Counter();
+
+        if (Strings.isFilled(className)) {
+            EntityDescriptor descriptor = mixing.getDescriptor(className.toUpperCase());
+            Class<? extends BaseEntity> type = descriptor.getType().asSubclass(BaseEntity.class);
+            Query<?, ?, ?> query = descriptor.getMapper().select(type).queryString(queryString);
+
+            sourceBuilder.append("/* Number of results: ").append(query.count()).append(" */\n\n");
+
+            ((Query<?, ?, ?>) query.limit(limit)).iterateAll(entity -> addEntity(sourceBuilder, entity, counter));
+        }
+
+        webContext.respondWith()
+                  .template("/templates/biz/model/db.html.pasta",
+                            getAvailableEntityTypes(),
+                            className,
+                            queryString,
+                            sourceBuilder.toString(),
+                            limit);
+    }
+
+    /**
+     * Provides suggestions for available database entities based on the query.
+     *
+     * @param webContext the current web context containing the query
+     */
+    @Permission(TenantUserManager.PERMISSION_SYSTEM_ADMINISTRATOR)
+    @Routed("/system/query/autocomplete")
+    public void entityTypeAutocomplete(WebContext webContext) {
+        AutocompleteHelper.handle(webContext, ((query, result) -> {
+            for (Class<?> type : getAvailableEntityTypes()) {
+                String displayableName = getDisplayableName(type);
+                if (displayableName.toLowerCase().contains(query.toLowerCase())) {
+                    result.accept(new AutocompleteHelper.Completion(type.getSimpleName(), displayableName));
+                }
+            }
+        }));
+    }
+
+    private <E extends BaseEntity<?>> void addEntity(StringBuilder sourceBuilder, E entity, Counter counter) {
+        sourceBuilder.append("/* ").append(counter.inc()).append(" */").append("\n");
+
+        Map<String, Object> properties = new TreeMap<>();
+        mixing.getDescriptor(entity.getClass()).getProperties().forEach(property -> {
+            String name = property.getName();
+            Object value = property.getValue(entity);
+
+            properties.put(name, value);
+        });
+
+        sourceBuilder.append(JSON.toJSONString(properties, true));
+        sourceBuilder.append("\n\n");
+    }
+
+    private List<Class<?>> getAvailableEntityTypes() {
+        return mixing.getDescriptors()
+                     .stream()
+                     .filter(descriptor -> MongoEntity.class.isAssignableFrom(descriptor.getType())
```

Comment:
As this would be more secure and easier to use than the SQL query tool, is there any reason not to include SQL Entities as well?

Reply:
Yes, because it's quite difficult to filter out unused SQL entities.

Reply:
there should be none